Matlab.h: warning: unknown option after ‘#pragma GCC diagnostic’ kind

Strangely, my GCC 4.8 does not recognize this pragma although it should according to [this OS thread](http://stackoverflow.com/questions/3378560/how-to-disable-gcc-warnings-for-a-few-lines-of-code):

```
In file included from /homes/as12312/Code/MIRTK/Modules/Numerics/include/mirtk/SparseMatrix.h:38:0,
                 from /homes/as12312/Code/MIRTK/Modules/PointSet/include/mirtk/EdgeTable.h:23,
                 from /homes/as12312/Code/MIRTK/Modules/PointSet/include/mirtk/RegisteredPointSet.h:28,
                 from /homes/as12312/Code/MIRTK/Modules/PointSet/include/mirtk/RegisteredSurface.h:23,
                 from /homes/as12312/Code/MIRTK/Applications/src/calculate-surface-spectrum.cc:25:
/homes/as12312/Code/MIRTK/Modules/Common/include/mirtk/Matlab.h:26:34: warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
   #pragma GCC diagnostic ignored "-Wunused-private-field"
```

```
> c++ --version
c++ (Ubuntu 4.8.4-2ubuntu1~14.04.1) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

One option is to use `-Wnopragmas` to disable all `#pragma` warnings, though that also means we wouldn't catch actual mistakes... maybe worth more investigation because I don't recall having had this warning before swapping out the CMake BASIS modules. But I cannot see how that changed matters...

--- a/Modules/Common/include/mirtk/Matlab.h
+++ b/Modules/Common/include/mirtk/Matlab.h
@@ -21,19 +21,14 @@
 #define MIRTK_Matlab_H
 
 // Disable "Private field 'dependency' is not used" warning
-#if defined(__GNUG__)
-  #pragma GCC diagnostic push
-  #pragma GCC diagnostic ignored "-Wunused-private-field"
-#elif defined __clang__
+#if defined __clang__
   #pragma clang diagnostic push
   #pragma clang diagnostic ignored "-Wunused-private-field"
 #endif
 // Include MATLAB Compiler Runtime header
 #include <mclmcrrt.h>
 // Enable warnings again
-#if defined(__GNUG__)
-  #pragma GCC diagnostic pop
-#elif defined __clang__
+#if defined __clang__
   #pragma clang diagnostic pop
 #endif
 

